File: Level-2/Sorting/BubbleSort.py

def swapfunction(x, y):
    temp = y
    y = x
    x = temp
    return x, y

def bubblesort(x):
    for i in range(len(x)-1):
        for i in range(len(x)-1):
            if x[i] > x[i+1]:
                x[i], x[i+1] = swapfunction(x[i], x[i+1])
                print(x)
            else:
                print(x)
        print("\n", x, "\n")
    print("\n", x)


bubblesort([5, 1, 4, 2, 8])

File: Level-2/Sorting/Bubblesort.cpp
#include <iostream>
#include <vector>
using namespace std;


int bubblesort(int arr[], int &size){

    for(int i; i < size-1; i++){
        for(int i; i < size-1; i++){
            if (arr[i]>arr[i+1]){
                int temp = arr[i];
                arr[i] = arr[i+1] ;
                arr[i] = temp;
            }
        }
    }
    return;
}


int main(){
    int rrr[] = {9,8,7,6,5,4,3};
    int size = sizeof(rrr)/sizeof(rrr[0]);
    bubblesort(rrr, size);

    for (int i; i < rrr.size(); i++){
        cout << rrr[i] << endl;
    }
    return 0;
}